### ISSUE TITLE
Decide how to proceed with S3 upload with colocation

### ISSUE THREAD

**Issue by MartinquaXD** [cowprotocol/services#1752]
We are currently uploading auctions to S3 in the `solver` pod. When we switch over to colocation this pod will cease to exist.
That means we either stop uploading auctions to S3 or we have to move that logic somewhere else.

Conceptually the natural choice would be to move that logic into the `autopilot`. But the `autopilot` does not really have the required data to generate the current format. The current format would be generated in the `legacy` solvers wrapper in the colocated world so very far removed from the `autopilot`.
So in conclusion the S3 upload still poses a bunch of questions.

**Comment by MartinquaXD** [comment]
@nlordell suggested to simply store the auction provided by the `autopilot`.

**Comment by fleupold** [comment]
We could do the same thing as we do for MEVBlocker and implement a dummy solver (builder in MEVBlocker's case) whose only job it is to upload the received instance and return an empty response. But maybe this is overkill and uploading it as part of the main binary is better.

**Comment by fleupold** [comment]
Notes from weekly:
- Should probably log the new version

@harisang is the solver team using the auction instance files at the moment? What for? Would not having those file be blocking production launch for co-location?

**Comment by harisang** [comment]
> Notes from weekly:
> 
>     * Should probably log the new version
> 
> 
> @harisang is the solver team using the auction instance files at the moment? What for? Would not having those file be blocking production launch for co-location?

I have to admit these are not used often. It has happened that for debugging reasons, we quickly check those since we used to log the "received json response" logs from solvers, and within those, orders were referenced with the index they were given within the auction json (and not their uid), so it was making things faster sometimes. I also feel some solvers from time to time use those for sanity checks and debugging.

Finally, and more importantly I guess, we currently use those for EBBO tests by invoking Quasimodo. This particular EBBO test has proved to be the most unstable (and not very useful) compared to the others we are running in the EBBO pod, but it is potentially something we might still want to use and improve.

Given the above, I would say this is definitely NOT blocking, especially if it is for just a few days/weeks that we are missing these files. But if we decide to drop them for good, we should communicate this to solvers, and within the solver team we should also rethink about some basic EBBO tests.

**Comment by fleupold** [comment]
The format that is currently used for the EBBO is the legacy `instance.json` format which we are hoping to phase out. It would be good to evaluate if the EBBO test and other longer term critical infrastructure tests can happen on the new auction format as exposed by https://api.cow.fi/docs/#/default/get_api_v1_auction

If not, and until then, we can create a standalone legacy solver whose only job is to upload the received `instance.json` file to AWS and return an empty solution. This could in theory be done using any programming language/infrastructure stack we want (all it needs to do is confirm to the legacy HTTP interface).

**Comment by fleupold** [comment]
@fhenneke please let me know your progress on this. We now have the S3 uploading logic in a shared rust crate and could potentially with relatively little effort add it to some legacy solver to upload both the new as well as the old instance file to a bucket.

I do think having a standalone solver is the cleaner way of doing things, but given everyones ressource constraints, I'd be ok to take that shortcut.

**Comment by fhenneke** [comment]
Due to vacation I did not get to implementing this and will not until the second week of January. If you can easily add it in the legacy solvers, it should be good enough.
Sorry for holding that issue up a bit.
